File: pset6/cash.py
get_input = True
while get_input:
    try:
        change = float(input("Change owed: "))
    except ValueError:
        pass
    else:
        if change > 0:
            get_input = False

counter = 0

change *= 100

quarter = True
while quarter:
    if change >= 25:
        change -= 25
        counter += 1
    else:
        quarter = False

dime = True
while dime:
    if change >= 10:
        change -= 10
        counter += 1
    else:
        dime = False

nickel = True
while nickel:
    if change >= 5:
        change -= 5
        counter += 1
    else:
        nickel = False

penny = True
while penny:
    if change >= 1:
        change -= 1
        counter += 1
    else:
        penny = False

print(counter)

File: pset1/cash.c
#include <stdio.h>
#include <stdbool.h>

int main(void)
{
	double amount = 12.48;
	
	bool coins[7] = {1, 1, 0, 1, 1, 1, 1}; // toonie, loonie, half-dollar, quarter, dime, nickel, penny
	int coinsused[7] = {0, 0, 0, 0, 0, 0, 0};
	
	while (amount >= 2)
	{
		coinsused[0]++;
		amount-=2;
	}
	while (amount >= 1)
	{
		coinsused[1]++;
		amount-=1;
	}
	while (amount >= 0.5)
	{
		coinsused[2]++;
		amount-=0.5;
	}
	while (amount >= 0.25)
	{
		coinsused[3]++;
		amount-=0.25;
	}
	while (amount >= 0.10)
	{
		coinsused[4]++;
		amount-=0.10;
	}
	while (amount >= 0.05)
	{
		coinsused[5]++;
		amount-=0.05;
	}
	while (amount >= 0.01)
	{
		coinsused[6]++;
		amount-=0.01;
	}
	
	for (int i = 0; i < 7; i++)
	{
		printf("%d\n", coinsused[i]);
	}
	
	return 0;
}